### PR TITLE
Fix #1610 unexpected field-error related to a well-formatted value example of a schema's boolean field using frictionless validate

### LINKED ISSUE #1610: Unexpected field-error for a boolean "example" with "trueValues" or "falseValues" properties
### Overview
Using frictionless validate I got this unvalid report of validation:
```
frictionless validate data.csv --schema schema.json
──────────────────────────────────────────────────────────────────────────────────────────────────────────── Dataset ─────────────────────────────────────────────────────────────────────────────────────────────────────────────
               dataset               
┏━━━━━━┳━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┓
┃ name ┃ type  ┃ path     ┃ status  ┃
┡━━━━━━╇━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━┩
│ data │ table │ data.csv │ INVALID │
└──────┴───────┴──────────┴─────────┘
───────────────────────────────────────────────────────────────────────────────────────────────────────────── Tables ─────────────────────────────────────────────────────────────────────────────────────────────────────────────
                                                 data                                                  
┏━━━━━━┳━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Row  ┃ Field ┃ Type        ┃ Message                                                                ┃
┡━━━━━━╇━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ None │ None  │ field-error │ Field is not valid: example value "no" for field "IsTrue" is not valid │
└──────┴───────┴─────────────┴────────────────────────────────────────────────────────────────────────┘
```
The error is related to the value given in the `example` property of the field 'IsTrue' of type `boolean` in the schema used for the validation, which is not valid. 

I was expected a valid report of validation: I was not expected an field-error on this value `example` which matches with the `falseValues` property value, as mentionned in the [`boolean` type of TableSchema documentation](https://specs.frictionlessdata.io/table-schema/#boolean).

### Resources used to reproduce the problem
- schema.json:
    ```
    {
      "$schema": "https://frictionlessdata.io/schemas/table-schema.json",
      "fields": [
        {
          "name": "IsTrue",
          "type": "boolean",
          "trueValues": ["yes"],
          "falseValues": ["no"],
          "example": "no"
        }]
    }

    ```
- data.csv:
    ```
    IsTrue
    no
    yes
    ```

### Note
Replacing in the schema field 'IsTrue' the optionnal property `example` with 'false' instead of 'no' solves the problem. The validation report is now valid with this new schema:
- schema.json:
    ```
    {
      "$schema": "https://frictionlessdata.io/schemas/table-schema.json",
      "fields": [
        {
          "name": "IsTrue",
          "type": "boolean",
          "trueValues": ["yes"],
          "falseValues": ["no"],
          "example": "false"
        }]
    }
    ```
- data.csv:
    ```
    IsTrue
    no
    yes
    ```

```
frictionless validate data.csv --schema schema.json
──────────────────────────────────────────────────────────────────────────────────────────────────────────── Dataset ─────────────────────────────────────────────────────────────────────────────────────────────────────────────
              dataset               
┏━━━━━━┳━━━━━━━┳━━━━━━━━━━┳━━━━━━━━┓
┃ name ┃ type  ┃ path     ┃ status ┃
┡━━━━━━╇━━━━━━━╇━━━━━━━━━━╇━━━━━━━━┩
│ data │ table │ data.csv │ VALID  │
└──────┴───────┴──────────┴────────┘

```

As specified in the [TableSchema boolean type documentation](https://specs.frictionlessdata.io/table-schema/#boolean) : 
*"The boolean field can be customised with these additional properties:
`trueValues: [ "true", "True", "TRUE", "1" ]
falseValues: [ "false", "False", "FALSE", "0" ]
`
`format`: no options (other than the default)."*

But, in this case the `falseValues` property seems to not be applied on the `example` property by frictionless and retruns an unexpected field-error in the validation report.

### PR DESCRIPTION
**Pull request by amelie-rondot**
- fixes #1610

This PR fixes the `field-error` which occured doing `frictionless validate data.csv --schema schema.json` with a  `TableSchema` `schema` containing a `BooleanField` customised with 'trueValues' or 'falseValues' and 'example' value , for example:
```
schema.json
----
{
  "$schema": "https://frictionlessdata.io/schemas/table-schema.json",
  "fields": [
    {
      "name": "IsTrue",
      "type": "boolean",
      "trueValues": ["yes"],
      "falseValues": ["no"],
      "example": "yes"
    }
  ]
}
```
```
data.csv
----
isTrue
yes
no
```

This PR add tests cases to ensure that example value is correctly evaluated according to customized 'trueValues' or 'falseValues', as expected.

### CODE REVIEW

**Comment by roll** [comment]
Thanks @amelie-rondot! Can you please fix the lining error?

**Comment by amelie-rondot** [comment]
@roll thanks a lot for reviewing!
I committed linting fixes.

File: frictionless/schema/field.py [hunk 1]
```diff
@@ -263,6 +280,12 @@ def metadata_validate(cls, descriptor: IDescriptor):  # type: ignore
             field = Class(
```
Comment (roll) [inline frictionless/schema/field.py:266]: I think we can just try `field = Class.from_descriptor(descriptor)` to make it work in general

File: frictionless/schema/field.py [hunk 2]
```diff
@@ -194,6 +196,21 @@ def value_writer(cell: Any):
...
+    @classmethod
+    def from_descriptor(  # type: ignore
+        cls,
+        descriptor: Union[IDescriptor, str],
+        **options: Any,
+    ) -> Self:
+        if isinstance(descriptor, dict):
+            if "trueValues" in descriptor.keys():
+                assert descriptor["type"] == "boolean"
+                descriptor["trueValues"] = descriptor["trueValues"]
```
Comment (roll) [inline frictionless/schema/field.py]: Not sure I understand this block of code =)

BTW as Table Schema is an open spec regarding custom properties we can't disable `true/falseValues` for other fields

**Review by roll — DISMISSED** [review]
Thanks!

I added a few comments

File: frictionless/schema/field.py (hunk 2 above)
Comment (amelie-rondot) [reply frictionless/schema/field.py]: Indeed, you're right. This block is now useless. I've just committed to remove it.

**Comment by amelie-rondot** [comment]
Thanks @roll for reviewing! I've committed updates.

**Comment by pierrecamilleri** [comment]
@roll 

The change requested have been taken into account and the PR is ready for review ! 

I noticed some linting errors that seem unrelated to this review, and that I corrected in the [other PR](https://github.com/frictionlessdata/frictionless-py/pull/1641) (should I have ?). I haven't corrected them in this one as it would mostly lead to merge conflicts. Tell me if you prefer me to do something about it.

File: frictionless/schema/field.py (hunk 1 above)
Comment (pierrecamilleri) [reply frictionless/schema/field.py:266]: It does not work with `from_descriptor`, it leads to a `RecursionError`.

**Comment by pierrecamilleri** [comment]
I merged the main branch, this PR is ready as well !

**Comment by pierrecamilleri** [comment]
@pdelboca, would it please be possible to prioritize the merges of this bug fix as well as PR #1633 ?
Validata currently needs to rely on a fork with these fixes to use frictionless v5, but we wouldn't want this situation to last...

**Review by pdelboca — APPROVED** [review]

**Review by pierrecamilleri — APPROVED** [review]

**Comment by pdelboca** [comment]
@pierrecamilleri I just merged #1674 . Could you please rebase and then merge when test are passing?

**Pull request merged**